test/distributed/_composable/test_compose.py

```diff
@@ -171,6 +179,37 @@ def test_checkpoint_fsdp_submodules_with_param_no_shard(self):
             self._test_parity,
         )
 
+    def test_composable_fsdp_replicate(self):
+        model = CompositeModel(device=torch.device("cpu"))
+        fully_shard(model.l1)
+        with self.assertRaisesRegex(
+            AssertionError, "Cannot apply .*replicate"
+        ):
+            replicate(model.l1)
+        replicate(model.l2)  # should not raise
+
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_in_replicate(self):
+        model = CompositeModel(device=torch.device("cuda"))
+        base_model = copy.deepcopy(model)
+        test_model = copy.deepcopy(model)
+        fully_shard(test_model.l1)
+        replicate(test_model)
+        self.run_subtests(
+            {
+                "base_model": [base_model],
+                "test_model": [test_model],
+                "x": [torch.randn(2, 100, device="cuda")],
+                "grad_to_none": [True, False],
+            },
+            self._test_parity,
+        )
+
+    @skip_if_lt_x_gpu(2)
+    def test_replicate_in_fsdp(self):
+        # TODO: to be added
```

Comment:
In general I feel like it's better to omit this and file an issue, or write out what the test might be like and use `assertRaises` for the parts that fail.